p2p/src/authenticated/lookup/actors/listener.rs

```diff
@@ -219,7 +219,11 @@ impl<E: Spawner + Clock + Network + Rng + CryptoRng + Metrics, C: Signer> Actor<
 
                 // Cleanup the rate limiters periodically
                 if accepted > CLEANUP_INTERVAL {
+                    // Clean up old entries and shrink the capacity of the rate limiters
+                    // to free up memory.
```

Comment:
These cleanups happen relatively infrequently and the RLs grow a lot larger than the others. It probably makes sense to clean up the capacity here and in discovery's listener actor due to a potential large volume of incoming requests being able to indefinitely expand memory otherwise..